src/libsyntax/ast.rs

```diff
@@ -1132,26 +1131,20 @@ pub enum ExprKind {
     Cast(P<Expr>, P<Ty>),
     /// A type ascription (e.g., `42: usize`).
     Type(P<Expr>, P<Ty>),
+    /// A `let pats = expr` pseudo-expression that only occurs in the scrutinee
+    /// of `if` / `while` expressions. (e.g., `if let 0 = x { .. }`).
```

Comment:
Is it necessary to use the "scrutinee" jargon here and in other places?
The thing is usually called if/while condition or header in human language.

Reply:
I wouldn't be able to guess exactly what "scrutinee" meant in an expression like this 👍

Reply:
Yeah, I know it's a Haskell term (or PL research term in general), but Centril's PRs are the first place where I actually encountered it.
Fun fact: google translate can't translate it.

Reply:
Done.

Reply:
There are several more instances.

Reply:
In the context of `let ps = e` or `match e` it would be incorrect to call `e` a "condition" because the condition is the whole `let ps = e` or the patterns in the `match` block. Scrutinee is the appropriate term of art for these and it is both used in the reference and in the compiler itself already in other places.

Reply:
>both used in the reference and in the compiler itself already in other places.

Ha, that's because you put them there (I've seen the PRs).

Reply:
In some places yes, e.g. the reference (with good reasons, because the reference should use common PLT terms), but not in others. See https://github.com/rust-lang/rust/search?q=scrutinee&unscoped_q=scrutinee (I only introduced a new function in `src/librustc_typeck/check/_match.rs`).

Reply:
I'd still personally prefer not to see this term of art, unless the code actually has to do something with PLT (like e.g. `match` checking/lowering, which is pretty algorithm-heavy and the implementation refers to papers).